sm/plugin_sm/src/plugin.rs

```diff
@@ -177,6 +181,57 @@ impl Plugin for ExternalPluginCommand {
         }
     }
 
+    async fn update_list(&self, updates: &Vec<SoftwareModuleUpdate>) -> Result<(), SoftwareError> {
+        let mut command = if let Some(sudo) = &self.sudo {
+            let mut command = Command::new(&sudo);
+            command.arg(&self.path);
+            command
+        } else {
+            Command::new(&self.path)
+        };
+        command
+            .arg(UPDATE_LIST)
+            .current_dir("/tmp")
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped());
+
+        let mut child = command.spawn()?;
+        let child_stdin = child.stdin.as_mut().ok_or_else(|| SoftwareError::IoError {
+            reason: "Plugin stdin unavailable".into(),
+        })?;
+
+        for update in updates {
+            let action = match update {
+                SoftwareModuleUpdate::Install { module } => {
+                    format!(
+                        "install {} {} {}\n",
+                        module.name,
+                        module.version.clone().map_or("".into(), |v| v),
+                        module.url.clone().map_or("".into(), |v| v)
+                    )
+                }
+
+                SoftwareModuleUpdate::Remove { module } => {
+                    format!(
+                        "remove {} {} {}\n",
+                        module.name,
+                        module.version.clone().map_or("".into(), |v| v),
+                        module.url.clone().map_or("".into(), |v| v)
```

Comment:
Does remove really needs to provide `url` as input?

Reply:
No it was done just to ease the implementation. We were free to change the spec if the impl of some aspect was unnecessarily hard for no added value. So, decided to pass empty URL for remove as well, just like SmartREST does.